sdks/java/testing/tpcds/src/main/java/org/apache/beam/sdk/tpcds/SqlTransformRunner.java

```diff
@@ -306,6 +327,68 @@ public static void runUsingSqlTransform(String[] args) throws Exception {
 
     executor.shutdown();
 
-    printExecutionSummary(completion, queryNames.length);
+    List<TpcdsRunResult> results = collectTpcdsResults(completion, queryNames.length);
+
+    if (tpcdsOptions.getExportSummaryToInfluxDB()) {
+      final long timestamp = start.getMillis() / 1000; // seconds
+      savePerfsToInfluxDB(tpcdsOptions, results, timestamp);
+    }
+
+    printExecutionSummary(results);
+  }
+
+  private static void savePerfsToInfluxDB(
+      final TpcdsOptions options, final List<TpcdsRunResult> results, final long timestamp) {
+    final InfluxDBSettings settings = getInfluxSettings(options);
+    final Map<String, String> tags = options.getInfluxTags();
+    final String runner = options.getRunner().getSimpleName();
+    final List<Map<String, Object>> schemaResults =
+        results.stream()
+            .map(
+                entry ->
+                    getResultsFromSchema(
+                        entry, timestamp, runner, produceMeasurement(options, entry)))
+            .collect(toList());
+    InfluxDBPublisher.publishTpcdsResults(schemaResults, settings, tags);
+  }
+
+  private static String produceMeasurement(final TpcdsOptions options, TpcdsRunResult entry) {
+    return String.format(
+        "%s_%s_%s_%s",
+        options.getBaseInfluxMeasurement(),
+        entry.getQueryName(),
+        entry.getDialect(),
+        entry.getDataSize());
```

Comment:
Wondering, particularly `dataSize` looks more like a tag than something making it an independent measurement. Also not sure about the others ...